addons/l10n_it_edi/models/account_move.py

```diff
@@ -476,7 +476,7 @@ def _l10n_it_edi_get_values(self, pdf_values=None):
 
         downpayment_lines = []
         # Prepare for '_dispatch_negative_lines'
-        for base_line in base_lines:
+        for base_line in list(base_lines):
```

Comment:
Making a copy of the `base_lines` with `list` is too subtle for me.
Let's be extra explicit and make one loop just for the partitioning.

```suggestion
        negative_downpayment_lines = []
        # [...]
        for base_line in base_lines:
        # [...]
            if not is_downpayment:
                # Negative lines linked to down payment should stay negative
                line = base_line['record']
                if line.price_subtotal < 0 and line._get_downpayment_lines():
                    negative_downpayment_lines.append(base_line)
        # [...]
        for negative_downpayment_line in negative_downpayment_lines:
            downpayment_lines.append(negative_downpayment_line)
            base_lines.remove(negative_downpayment_line)
```